tests/src/test/java/alluxio/server/ft/journal/TableMasterJournalIntegrationTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.server.ft.journal;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+
+import alluxio.master.LocalAlluxioCluster;
+import alluxio.master.table.Table;
+import alluxio.master.table.TableMaster;
+import alluxio.master.table.TestDatabase;
+import alluxio.master.table.TestUdbFactory;
+import alluxio.testutils.LocalAlluxioClusterResource;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Integration tests for table master functionality.
+ */
+public class TableMasterJournalIntegrationTest {
+  @Rule
+  public LocalAlluxioClusterResource mClusterResource =
+      new LocalAlluxioClusterResource.Builder().setStartCluster(false)
+          .build();
+
+  private static final String DB_NAME = TestDatabase.DB_NAME;
+
+  @Before
+  public void before() throws Exception {
+    TestDatabase.genTable(1);
+  }
+
+  @Test
+  public void journalAttachandDetachDb() throws Exception {
```

Comment:
can we split up the tests so there is a test for "attach" (attach , restart, check), a test for "detach" (attach, detach, restart, check), and a test for layout changes?

Reply:
split into attach and detach
@chcn will add the test for layout changes